src/hawkmoth/__init__.py

```diff
@@ -160,6 +160,7 @@ class _AutoDocDirective(_AutoBaseDirective):
     optional_arguments = 100   # arbitrary limit
 
```

Comment:
You can always do `list(_get_filenames())` where needed and allow for a marginally better performance and semantics in the cases where it's not needed.

Though this is a drive by comment and it's very possible this is still the way to go. It was the commit message that caught my eye.